src/ripple/protocol/STAmount.h

```diff
@@ -104,6 +104,32 @@ class STAmount final
 
     STAmount (Issue const& issue, int mantissa, int exponent = 0);
 
+    std::size_t
+    size_of() const override
+    {
+        return sizeof(*this);
+    }
+
+    STBase*
+    copy (std::size_t n, void* buf) const override
```

Comment:
Modulo macros, I know of no way to eliminate the need to define `copy` and `move` for each class.  And I would rather see that repetition than a macro.  The macro solution would not be readable.